src/Debugger/Impl/ExecutionTracing/RBreakpoint.cs

```diff
@@ -10,19 +10,20 @@
 using Microsoft.R.Host.Client;
 using static System.FormattableString;
 
-namespace Microsoft.R.Debugger {
-    /// <summary>
-    /// A breakpoint in R code.
-    /// </summary>
-    public sealed class DebugBreakpoint {
-        public DebugSession Session { get; }
-        public DebugSourceLocation Location { get; }
+namespace Microsoft.R.ExecutionTracing {
+    internal sealed class RBreakpoint : IRBreakpoint {
+        private readonly RExecutionTracer _tracer;
+
+        public IRExecutionTracer Tracer => _tracer;
```

Comment:
That doesn't work, because return types don't match - and _tracer has to be RExecutionTracer rather than IRExecutionTracer, so that RemoveBreakpoint can be called on it.

---
In reply to: [62136948](https://github.com/Microsoft/RTVS/pull/1731#discussion_r62136948) [](ancestors = 62136948)